code/modules/robotics/bot/floorbot.dm

```diff
@@ -474,10 +474,7 @@
 	if(src.color_overlay)
 		A.UpdateOverlays(image(A.icon, icon_state = src.color_overlay), "coloroverlay")
 		A.color_overlay = src.color_overlay
-	if (user.r_hand == src || user.l_hand == src)
-		A.set_loc(user.loc)
-	else
-		A.set_loc(src.loc)
+	A.set_loc(get_turf(src))
```

Comment:
I know I said `get_turf` was the way to do this on discord, but the more I think about it, just always using `user.loc` seems better- in theory you should be able to build floorbots in closets/pods/disposal chutes/etcetc without them immediately popping out onto the floor.